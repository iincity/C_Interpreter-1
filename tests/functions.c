
#include <stdio.h>

int a = 1;

void func1(void)
{
	printf("%d\n", a);
	int a = 2;
	printf("%d\n", a);
	int local = 10 * a;
	printf("%d\n", local);
}

void func2(int a, int b, int c)
{
	printf("%d\n", a);
	printf("%d\n", b);
	printf("%d\n", c);

	int d = 12;
	{
		printf("%d\n", d);
		printf("%d\n", a);
		int a = 15;
		printf("%d\n", a);
		{
			printf("%d\n", d);
			printf("%d\n", a);
			printf("%d\n", b);

			int b = 100;
			printf("%d\n", b);
			printf("%d\n", c);
			{
				printf("%d\n", c);
			}

		}
		int c = 5;
		printf("%d\n", c);

	}
}

int fibbonaci(int n)
{
	printf("%d\n", n);
	if (n < 3) {
		return 1;
	}
	int a = n + 2;
	printf("%d\n", a);
	return fibbonaci(n-2) + fibbonaci(n-1);
}


int main()
{
	int local = 54;
	printf("%d\n", local);
	func1();
	printf("%d\n", a);
	func2(10, local /= 3, local - a*20);
	printf("%d\n", local);

	local = fibbonaci(4);
	printf("%d\n", local);

	return 0;
}


