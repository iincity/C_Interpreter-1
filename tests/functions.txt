
int a = 1;

void func1(void)
{
	print a;
	int a = 2;
	print a;
	int local = 10 * a;
	print local;
}


void func2(int a, int b, int c)
{
	print a;
	print b;
	print c;

	int d = 12;
	{
		print d;
		print a;
		int a = 15;
		print a;
		{
			print d;
			print a;
			print b;

			int b = 100;
			print b;
			print c;
			{
				print c;
			}

		}
		int c = 5;
		print c;

	}
}

int fibbonaci(int n)
{
	print n;
	if (n < 3) {
		return 1;
	}
	int a = n + 2; //removing shaves off 52 seconds for n = 40
	print a;
	return fibbonaci(n-2) + fibbonaci(n-1);
}



int main()
{
	int local = 54;
	print local;
	func1();
	print a;
	func2(10, local /= 3, local - a*20);
	print local;

	local = fibbonaci(4);
	print local;

	return 0;
}
