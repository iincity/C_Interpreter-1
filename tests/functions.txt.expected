54
1
2
20
1
10
18
34
12
10
15
12
15
18
100
34
34
5
18
